processing/src/main/java/org/apache/carbondata/processing/loading/sort/unsafe/merger/UnsafeIntermediateMerger.java

```diff
@@ -111,18 +108,15 @@ public void addFileToMerge(File sortTempFile) {
   }
 
   public void startFileMergingIfPossible() {
-    File[] fileList = null;
-    synchronized (lockObject) {
-      if (procFiles.size() >= parameters.getNumberOfIntermediateFileToBeMerged()) {
+    File[] fileList;
+    if (procFiles.size() >= parameters.getNumberOfIntermediateFileToBeMerged()) {
```

Comment:
@kumarvishal09 Please check once, in my view here double check locking is needed, other wise the thread waiting to acquire the lock, will enter the synchronized block and will end up doing intermediate merging with **0 or less than configured number of files "carbon.sort.intermediate.files.limit**".

Reply:
@mohammadshahidkhan Yes you are right but in this case UnsafeSortDataRow processing will be slower as it will read/ sort and write so chances of above condition is negligible, because of this double check is not added here

Reply:
ok